src/v/cluster/rm_stm.cc

```diff
@@ -1645,11 +1645,17 @@ model::offset rm_stm::last_stable_offset() {
     }
 
     auto last_visible_index = _c->last_visible_index();
```

Comment:
just a quick question here, are we guaranteed that `last_visible_index` is applied ?

Reply:
by default raft only applies entries up to `_committed_index`

Reply:
I think @mmaslankaprv is right and we should futurize last_stable_offset and sync before use or switch to _committed_index; the latter is probably better because we'll add read replicas eventually and sync work only for a leader

Reply:
created an issue to follow https://github.com/redpanda-data/redpanda/issues/7232